test/test_jit.py

```diff
@@ -10454,6 +10454,30 @@ def forward2(self, v):
             o2 = m.forward2(i)
             self.assertEqual(o2, v)
 
+    def test_script_sequential_sliced_iteration(self):
+        class seq_mod(nn.Module):
+            def __init__(self):
+                super(seq_mod, self).__init__()
+                self.layers = [nn.ReLU(), nn.ReLU(), nn.ReLU()]
+                self.layers = nn.Sequential(*self.layers)
+
+            def forward(self, input):
+                x = layers[0].forward(input)
```

Comment:
mind adding a couple tests for `nn.ModuleList` (which should also work) and for an optionally present end step?